web-app/app/tricorder/page.tsx

```diff
@@ -1,97 +1,87 @@
 "use client";
 
-import { useEffect, useState } from "react";
+import { useEffect, useRef, useState } from "react";
 import Counter from "./Counter";
-import { Configure } from "./Configure";
+import { Config, ConfigurationDisplay, Configure } from "./Configure";
 
 function Tricorder(props: any) {
-  const scanForLifeforms = useState<any>();
+  const [config, setConfig] = useState<Config>({});
+  const [isScanningForLifeForms, setIsScanningForLifeforms] =
+    useState<boolean>();
+  const [foundLifeforms, setFoundLifeForms] = useState<number[]>([]);
 
-  var isScanningForLifeForms = scanForLifeforms[0];
-  var setScanForLifeFormsToTrue = () => scanForLifeforms[1](true);
-  var setScanForLifeFormsToFalse = () => scanForLifeforms[1](false);
+  const intervalRef = useRef<NodeJS.Timeout>();
 
-  let [foundLifeforms, setFoundLifeForms] = useState<any>([]);
+  const soughtLifeformCount = 47;
 
   useEffect(() => {
-    setInterval(() => {
-      foundLifeforms.push(Math.random());
-    }, 1000);
-  });
+    const multipliers =
+      (config.antimatter ? 0.5 : 1) *
+      (config.slingshot ? 0.5 : 1) *
+      (config.stembolts ? 0.5 : 1);
+    const interval = setInterval(() => {
+      if (isScanningForLifeForms) {
+        setFoundLifeForms((lifeforms) => {
+          if (lifeforms.length === soughtLifeformCount && intervalRef.current) {
+            clearInterval(intervalRef.current);
+          }
+          return lifeforms.length < soughtLifeformCount
+            ? [...lifeforms, Math.random()]
+            : lifeforms;
+        });
+      }
+    }, 1000 * multipliers);
+    intervalRef.current = interval;
+    return () => {
+      clearInterval(interval);
+    };
+  }, [isScanningForLifeForms, config, soughtLifeformCount]);
 
+  /**
+   * Update the page title with the number of found lifeforms when the array chnges
+   */
   useEffect(() => {
     document.title = "Found lifeforms: " + foundLifeforms.length;
-  });
-
-  const [titleSet, setTitleSet] = useState(false);
+  }, [foundLifeforms]);
 
+  /**
+   * Set the config in local storage
+   */
   useEffect(() => {
-    if (titleSet === false) {
-      document.title = "Found lifeforms: " + foundLifeforms.length;
-    }
-  }, [titleSet, foundLifeforms.length]);
-
-  const [config, setConfig] = useState<{
-    antimatter?: boolean;
-    slingshot?: boolean;
-    stembolts?: boolean;
-  }>();
-
-  const CheckConfigurationComponent = (tricorderConfig: typeof config) => {
```

Comment:
We should always define our components separately — not nested. This is hard to read and much more difficult to test.